draft-yasskin-http-origin-signed-responses.md

```diff
@@ -655,33 +671,52 @@ authoritative for the PUSH_PROMISE's authority.
       trusted logs. ({{!RFC6962}})
 1. Return "valid".
 
+### Open Questions ### {#oq-cross-origin-push}
+
+Is it right that "validityUrl" is required to be same-origin with the exchange?
+This allows the mitigation against downgrades in {{seccons-downgrades}}, but
+prohibits intermediates from providing a cache of the validity information.
+
 # Security considerations
 
+## Confidential data ## {#seccons-confidentiality}
+
 Authors MUST NOT include confidential information in a signed response that an
 untrusted intermediate could forward, since the response is only signed and not
 encrypted. Intermediates can read the content.
 
+## Off-path attackers ## {#seccons-off-path}
+
 Relaxing the requirement to consult DNS when determining authority for an origin
 means that an attacker who possesses a valid certificate no longer needs to be
 on-path to redirect traffic to them; instead of modifying DNS, they need only
 convince the user to visit another Web site in order to serve responses signed
-as the target. This consideration and mitigations for it are shared by
-{{?I-D.ietf-httpbis-origin-frame}}.
+as the target. This consideration and mitigations for it are shared by the
+combination of {{?I-D.ietf-httpbis-origin-frame}} and
+{{?I-D.ietf-httpbis-http2-secondary-certs}}.
+
+## Downgrades ## {#seccons-downgrades}
 
 Signing a bad response can affect more users than simply serving a bad response,
 since a served response will only affect users who make a request while the bad
 version is live, while an attacker can forward a signed response until its
 signature expires. Authors should consider shorter signature expiration times
 than they use for cache expiration times.
 
+Clients MAY also check the ["validityUrl"](#signature-validityurl) of an
+exchange more often than the signature's expiration would require. Doing so
+provides a TLS guarantee that the exchange isn't out of date.
```

Comment:
This guarantee is provided iff you have the same-origin assertion from line 647. The open question (in line 674) makes it unclear whether this would be true, but if that's removed, then it could be that validityUrl is served by a different entity, and thus does not provide a TLS guarantee.

Reply:
Also, given validityUrls are unsigned and don't *seem* to be restricted to https:// in any of the normative requirements, I'm not sure that is inherently true, so much as assumed true?